Makefile

```diff
@@ -0,0 +1,58 @@
+# This is used by the image building script referenced below. Normally it just takes the directory name but in this
+# case we want it to be called something else.
+IMAGE_NAME=flytekit-python-template
+VERSION=$(shell ./version.sh)
+
+define PIP_COMPILE
+pip-compile $(1) --upgrade --verbose
+endef
+
+# If the REGISTRY environment variable has been set, that means the image name will not just be tagged as
+#   flytecookbook:<sha> but rather,
+#   docker.io/lyft/flytecookbook:<sha> or whatever your REGISTRY is.
+ifneq ($(origin REGISTRY), undefined)
+	FULL_IMAGE_NAME = ${REGISTRY}/${IMAGE_NAME}
+else
+	FULL_IMAGE_NAME = ${IMAGE_NAME}
+endif
+
+# The Flyte project that we want to register under
+PROJECT=flyteexamples
+# If you want to create a new project, in an environment with flytekit installed run the following:
+# flyte-cli register-project -h localhost:30081 -i - myflyteproject --name "My Flyte Project" \
+#      --description "My very first project getting started on Flyte"
+
+.SILENT: help
+.PHONY: help
+help:
+	echo Available recipes:
+	cat $(MAKEFILE_LIST) | grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' | awk 'BEGIN { FS = ":.*?## " } { cnt++; a[cnt] = $$1; b[cnt] = $$2; if (length($$1) > max) max = length($$1) } END { for (i = 1; i <= cnt; i++) printf "  $(shell tput setaf 6)%-*s$(shell tput setaf 0) %s\n", max, a[i], b[i] }'
+	tput sgr0
+
+.PHONY: debug
+debug:
+	echo "IMAGE NAME ${IMAGE_NAME}"
+	echo "FULL IMAGE NAME ${FULL_IMAGE_NAME}"
+	echo "VERSION TAG ${VERSION}"
+	echo "REGISTRY ${REGISTRY}"
+
+.PHONY: docker_build
+docker_build:
+	NOPUSH=1 IMAGE_NAME=${IMAGE_NAME} flytekit_build_image.sh ./Dockerfile ${PREFIX}
```

Comment:
should we start moving away from this script and just use a raw docker command?

Reply:
sure? why did we add the script in the first place though if it's omittable?

Reply:
just to add the ARG that you have in the dockerfile - @wild-endeavor I would think do we need the ARG?

Reply:
@kumare3 there's some other stuff in there about checking for uncommitted git changes, and @wild-endeavor adds  registry stuff, and logging in for you, etc

can we keep it for now since it's useful?

Reply:
sure